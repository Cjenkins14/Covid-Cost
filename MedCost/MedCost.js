import React from 'react';

function MedCost(count) {
    let totalCost = parseInt(count.count) * 30000;
    let string = totalCost.toString();
    let result = string.replace(/(\d)(?=(\d\d\d)+(?!\d))/g, "$1,");

    return (
        <>
        <h3>New hospital patients (daily): {count.count}</h3>
        <h3>Medical costs (daily): ${result}</h3>
        </>
    )
}

export default MedCost;