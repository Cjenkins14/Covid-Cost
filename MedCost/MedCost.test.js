import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MedCost from './MedCost';

describe('medcost component', () => {
    const props = {
        count: 1
    }

    it('renders comp by default', () => {
        const wrapper = shallow(<MedCost/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders the comp given props', () => {
        const wrapper = shallow(<MedCost {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})