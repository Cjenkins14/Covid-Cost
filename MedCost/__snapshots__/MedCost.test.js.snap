// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`medcost component renders comp by default 1`] = `
<Fragment>
  <h3>
    New hospital patients (daily): 
  </h3>
  <h3>
    Medical costs (daily): $
    NaN
  </h3>
</Fragment>
`;

exports[`medcost component renders the comp given props 1`] = `
<Fragment>
  <h3>
    New hospital patients (daily): 
    1
  </h3>
  <h3>
    Medical costs (daily): $
    30,000
  </h3>
</Fragment>
`;
